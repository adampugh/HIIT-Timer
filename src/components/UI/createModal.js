import React, { Component } from "react";
import { connect } from "react-redux";
import Modal from "react-modal";

import * as actions from "../../store/actions/actions";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      }
}

export class CreateModal extends Component { 
    state = {
        workoutUrl: ""
    }

    updateUrl = (e) => {
        this.setState({
            workoutUrl: e.target.value
        });
    }

    handleOnClick = () => {
        let workout = {
            title: this.state.workoutUrl,
            totalTime: 0,
            exercises: [],
            index: this.props.workoutIndex
        };
        this.props.startAddWorkout(workout);
    }

    render() {

        return (
            <Modal 
                isOpen={this.props.isOpen}
                contentLabel="create workout modal"
                appElement={document.getElementById('root')} 
                onRequestClose={this.props.handleCloseModal}
                style={customStyles}
                closeTimeoutMS={200}
                className="modal"
            >
            <div className="modal__box">
                <div className="modal__header">
                    <div></div>
                    <div><h3>Create Workout</h3></div>
                    <div className="modal__close">
                        <h3>
                            <span onClick={this.props.handleCloseModal}>
                                <i className="fas fa-times"></i>
                            </span>
                        </h3>
                    </div>
                </div>
                <hr />
                <div className="modal--create">
                    <label>Name</label>
                    <input type="text" maxLength="20" onChange={(e) => this.updateUrl(e)}/>
                    
                        <button 
                            onClick={this.handleOnClick} 
                            disabled={this.state.workoutUrl.length < 1} 
                            className="modal__button">
                            Create Workout
                        </button>
                   
                </div>
            </div>
        </Modal>
        )
    }
}    

const mapDispatchToProps = dispatch => {
    return {
        startAddWorkout: (workout) => dispatch(actions.startAddWorkout(workout))
    }
}


export default connect(null, mapDispatchToProps)(CreateModal);