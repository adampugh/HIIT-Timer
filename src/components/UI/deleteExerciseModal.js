import React, { Component } from "react";
import Modal from "react-modal";
import { connect } from "react-redux";

import * as actions from "../../store/actions/actions";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      },
      content : {
      }
}

// pass in index and dispatch action
class deleteExerciseModal extends Component {
    handleDelete(index, workoutId) {
        this.props.deleteExercise(index, workoutId);
        this.props.handleCloseModal();
    }


    render() {
        return (
            <Modal 
                isOpen={this.props.isOpen}
                contentLabel="login modal"
                appElement={document.getElementById('root')} 
                onRequestClose={this.props.handleCloseModal}
                style={customStyles}
                closeTimeoutMS={200}
                className="modal"
            >
                <div className="modal__box">
                    <div className="modal__header">
                        <div></div>
                        <div><h3>Delete {this.props.modalTitle}</h3></div>
                        <div className="modal__close">
                            <h3>
                                <span onClick={this.props.handleCloseModal}>
                                    <i className="fas fa-times"></i>
                                </span>
                            </h3>
                        </div>
                    </div>
                    <hr />
                    <p>Do you want to delete this {this.props.modalTitle.toLowerCase()}?</p>
                    <button className="modal__login-button" onClick={() => this.handleDelete(this.props.exerciseIndex, this.props.workoutId)} >
                        <span>Yes</span>
                    </button>
                    <button className="modal__login-button" onClick={this.props.handleCloseModal}>
                        <span>No</span>
                    </button>
                </div>
            </Modal>
        );
    }
};

const mapDispatchToProps = dispatch => {
    return {
        deleteExercise: (index, id) => dispatch(actions.deleteExercise(index, id))
    }
}

export default connect(null, mapDispatchToProps)(deleteExerciseModal);