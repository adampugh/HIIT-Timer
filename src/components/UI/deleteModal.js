import React from "react";
import Modal from "react-modal";



const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      },
      content : {
      }
}

const loginModal = (props) => (
    <Modal 
        isOpen={props.isOpen}
        contentLabel="login modal"
        appElement={document.getElementById('root')} 
        onRequestClose={props.handleCloseModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal"
    >
        <div className="modal__box">
            <div className="modal__header">
                <div></div>
                <div><h3>Delete {props.modalTitle}</h3></div>
                <div className="modal__close">
                    <h3>
                        <span onClick={props.handleCloseModal}>
                            <i className="fas fa-times"></i>
                        </span>
                    </h3>
                </div>
            </div>
            <hr />
            <p>Do you want to delete this {props.modalTitle.toLowerCase()}?</p>
            <button className="modal__login-button">

                <span>Yes</span>
            </button>
            <button className="modal__login-button">

                <span>No</span>
            </button>
        </div>
    </Modal>
);

export default loginModal;