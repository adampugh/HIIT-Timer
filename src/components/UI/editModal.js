import React, { Component } from "react";
import Modal from "react-modal";
import { withFormik, Field, Form } from "formik";
import Yup from "yup";


const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      }
};

const editModal = (props) => (
    <Modal 
        isOpen={props.isOpen}
        contentLabel="add exercise modal"
        appElement={document.getElementById('root')} 
        onRequestClose={props.handleCloseModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal modal--edit"
    >
        <div className="modal__box">
            <div className="modal__header">
                <div></div>
                <div><h3>Add Exercise</h3></div>
                <div className="modal__close">
                    <h3>
                        <span onClick={props.handleCloseModal}>
                            <i className="fas fa-times"></i>
                        </span>
                    </h3>
                </div>
            </div>
            <hr />
            <div className="modal--edit">
                <Form>
                    <label>Name</label>
                    <Field type="text" maxLength="20" name="exerciseName" />
                    <div className="exerciseGrid">
                        <div className="exerciseGrid__timeCol">
                            <label>Time</label>
                            <Field className="modal--edit__numberInput1" type="number" max="60" name="exerciseMinutes" />
                            :
                            <Field className="modal--edit__numberInput2" type="number" max="59" name="exerciseSeconds" />
                        </div>
                        <div className="exerciseGrid__colorCol">
                            <label>Color</label>
                            <Field component="select" name="exerciseColor" style={{backgroundColor: props.values.exerciseColor}}>
                                <option value="#40b3e1"></option>
                                <option value="#acff82"></option>
                                <option value="#fedc45"></option>
                                <option value="#fec345"></option>
                                <option value="#fe6145"></option>
                                <option value="#b145fe"></option>
                            </Field>
                        </div>
                    </div>
                    
                        <div>
                            <label className="modal--edit__breakLabel">Break</label>
                        </div>
                        <div className="modal--edit__checkboxDiv">
                            <Field id="switch" className="modal--edit__checkbox" type="checkbox" name="breakIncluded" checked={props.values.breakIncluded} />
                            <label for="switch">Toggle</label>
                        </div>
                    
                    { props.values.breakIncluded &&
                        <div className="exerciseGrid">
                            <div className="exerciseGrid__timeCol">
                                <label>Time</label>
                                <Field className="modal--edit__numberInput1" type="number" max="60" name="breakMinutes" />
                                :
                                <Field className="modal--edit__numberInput2" type="number" max="59" name="breakSeconds" />
                            </div>
                            <div className="exerciseGrid__colorCol">
                                <label>Color</label>
                                <Field component="select" name="breakColor" style={{backgroundColor: props.values.breakColor}}>
                                    <option value="#40b3e1"></option>
                                    <option value="#acff82"></option>
                                    <option value="#fedc45"></option>
                                    <option value="#fec345"></option>
                                    <option value="#fe6145"></option>
                                    <option value="#b145fe"></option>
                                </Field>
                            </div>
                        </div>
                    }
                    <button className="modal__button">Add Exercise</button>
                </Form>
            </div>
        </div>
    </Modal>
);

const formikEditModal = withFormik({
    mapPropsToValues() {
        return {
            exerciseName: "",
            exerciseMinutes: 0,
            exerciseSeconds: 0,
            exerciseColor: "",
            breakIncluded: true,
            breakName: "Break",
            breakMinutes: 0,
            breakSeconds: 0,
            breakColor: "grey"
        }
    },
    handleSubmit(values) {
        console.log(values);
    }
})(editModal);

export default formikEditModal;
// <div className="modal--edit__breakInclude">
// <div>
//     <label className="modal--edit__breakLabel">Break</label>
// </div>
// <div className="modal--edit__checkboxDiv">
//     <Field id="switch" className="modal--edit__checkbox" type="checkbox" name="breakIncluded" checked={props.values.breakIncluded} />
//     <label for="switch">Toggle</label>
// </div>
// </div>