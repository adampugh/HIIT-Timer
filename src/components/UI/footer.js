import React from "react";

import SocialLinks from "./socialLinks";

const footer = () => (
    <footer>
        <div className="container">
            <h1>HIIT Timer</h1>
            <SocialLinks />
            <h3>&copy; 2018 Design by <a href="https://adampugh.github.io/kokonoka/" target="_blank" rel="noopener noreferrer">Kokonoka </a></h3>
        </div>
    </footer>
);

export default footer;