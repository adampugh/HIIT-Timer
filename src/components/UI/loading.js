import React from "react";

import dumbbellIcon from "../../assets/photos/dumbbell-icon.png";


const loading = () => (
    <div className="loading">
        <img className="workoutsBanner__image" src={dumbbellIcon} alt="arm icon" />
        <h2>. . . Loading</h2>
    </div>
);

export default loading;