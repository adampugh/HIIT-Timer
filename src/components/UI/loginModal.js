import React from "react";
import Modal from "react-modal";

import GoogleIcon from "../../assets/photos/googlelogo.jpg";
import FacebookIcon from "../../assets/photos/facebooklogo.jpg";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      },
      content : {
      }
}

const loginModal = (props) => (
    <Modal 
        isOpen={props.isOpen}
        contentLabel="login modal"
        appElement={document.getElementById('root')} 
        onRequestClose={props.handleCloseModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal"
    >
        <div className="modal__box">
            <div className="modal__header">
                <div></div>
                <div><h3>Log In</h3></div>
                <div className="modal__close">
                    <h3>
                        <span onClick={props.handleCloseModal}>
                            <i className="fas fa-times"></i>
                        </span>
                    </h3>
                </div>
            </div>
            <hr />

            <button className="modal__login-button">
                <img src={GoogleIcon} alt="google logo" />
                <span>Sign in with Google</span>
            </button>
            <button className="modal__login-button">
                <img src={FacebookIcon} alt="facebook logo" />
                <span>Sign in with Facebook</span>
            </button>
        </div>
    </Modal>
);

export default loginModal;