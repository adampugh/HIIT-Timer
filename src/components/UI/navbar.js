import React from "react";
import { NavLink } from "react-router-dom";

const navbar = (props) => (
    <nav className="navbar">
        <div className="container">
            <div className="navbar__items">
                <h2>HIIT Timer</h2>
                <div className="navbar__links">
                    {props.links.map(item => 
                        <NavLink 
                            key={item}
                            to={item.split(" ").join("").toLowerCase()}
                            activeClassName={"navbar__links--active"}>
                                {item}
                        </NavLink>
                    )}
                </div>
            </div>
        </div>
    </nav>
);

export default navbar;