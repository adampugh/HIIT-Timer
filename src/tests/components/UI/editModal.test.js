import React from "react";
import { shallow } from "enzyme"; 

import { FormikEditModal, EditModal } from "../../../components/UI/editModal";
import workoutState from "../../fixtures/workouts";

test("should render EditModal correctly", () => {
    const wrapper = shallow(<EditModal 
        errors={{
            exerciseName: "",
            exerciseColor: ""
        }}
        values={{
            exerciseColor: "blue"
        }} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render FormikEditModal correctly", () => {
    const wrapper = shallow(<FormikEditModal />);
    expect(wrapper).toMatchSnapshot();
});

test("should call startAddExercise for exercise and break in FormikEditModal", () => {
    const values = {
        exerciseName: "squats",
        exerciseMinutes: 1,
        exerciseSeconds: 10,
        exerciseColor: "#40b3e1",
        breakIncluded: true,
        breakName: "Break",
        breakMinutes: 1,
        breakSeconds: 10,
        breakColor: "#b145fe"
    }
    const startAddExerciseSpy = jest.fn();
    const wrapper = shallow(<FormikEditModal 
        startAddExercise={startAddExerciseSpy} 
        values={values} 
        workoutId={workoutState.workouts[0].id} 
        handleCloseModal={() => {}} />);
    wrapper.instance().handleSubmit(values, wrapper.props);
    expect(startAddExerciseSpy).toHaveBeenCalledWith({
        name: "squats",
        color: "#40b3e1",
        time: 70
    }, workoutState.workouts[0].id);
    expect(startAddExerciseSpy).toHaveBeenLastCalledWith({
        name: "Break",
        color: "#b145fe",
        time: 70
    }, workoutState.workouts[0].id);
    expect(startAddExerciseSpy).toHaveBeenCalledTimes(2);
});

test("should call startAddExercise for exercise only when breakIncluded is false in FormikEditModal", () => {
    const values = {
        exerciseName: "squats",
        exerciseMinutes: 1,
        exerciseSeconds: 10,
        exerciseColor: "#40b3e1",
        breakIncluded: false,
        breakName: "Break",
        breakMinutes: 1,
        breakSeconds: 10,
        breakColor: "#b145fe"
    }
    const startAddExerciseSpy = jest.fn();
    const wrapper = shallow(<FormikEditModal 
        startAddExercise={startAddExerciseSpy} 
        values={values} 
        workoutId={workoutState.workouts[0].id} 
        handleCloseModal={() => {}} />);
    wrapper.instance().handleSubmit(values, wrapper.props);
    expect(startAddExerciseSpy).toHaveBeenLastCalledWith({
        name: "squats",
        color: "#40b3e1",
        time: 70
    }, workoutState.workouts[0].id);
    expect(startAddExerciseSpy).toHaveBeenCalledTimes(1);
});