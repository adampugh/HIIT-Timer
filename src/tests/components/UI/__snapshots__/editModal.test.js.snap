// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditModal correctly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal modal--edit"
  closeTimeoutMS={200}
  contentLabel="add exercise modal"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "overlay": Object {
        "background": "rgba(0, 0, 0, 0.6)",
        "zIndex": 2000,
      },
    }
  }
>
  <div
    className="modal__box"
  >
    <div
      className="modal__header"
    >
      <div />
      <div>
        <h3
          onClick={[Function]}
        >
          Add Exercise
        </h3>
      </div>
      <div
        className="modal__close"
      >
        <h3>
          <span>
            <i
              className="fas fa-times"
            />
          </span>
        </h3>
      </div>
    </div>
    <hr />
    <div
      className="modal--edit"
    >
      <Form>
        <label>
          Name
        </label>
        <Field
          maxLength="20"
          name="exerciseName"
          type="text"
        />
        <div
          className="exerciseGrid"
        >
          <div
            className="exerciseGrid__timeCol"
          >
            <label>
              Time
            </label>
            <Field
              className="modal--edit__numberInput1"
              component="select"
              max="59"
              name="exerciseMinutes"
              type="text"
            >
              <option
                key="exerciseMin00"
                value="00"
              >
                00
              </option>
              <option
                key="exerciseMin01"
                value="01"
              >
                01
              </option>
              <option
                key="exerciseMin02"
                value="02"
              >
                02
              </option>
              <option
                key="exerciseMin03"
                value="03"
              >
                03
              </option>
              <option
                key="exerciseMin04"
                value="04"
              >
                04
              </option>
              <option
                key="exerciseMin05"
                value="05"
              >
                05
              </option>
              <option
                key="exerciseMin06"
                value="06"
              >
                06
              </option>
              <option
                key="exerciseMin07"
                value="07"
              >
                07
              </option>
              <option
                key="exerciseMin08"
                value="08"
              >
                08
              </option>
              <option
                key="exerciseMin09"
                value="09"
              >
                09
              </option>
              <option
                key="exerciseMin10"
                value="10"
              >
                10
              </option>
              <option
                key="exerciseMin11"
                value="11"
              >
                11
              </option>
              <option
                key="exerciseMin12"
                value="12"
              >
                12
              </option>
              <option
                key="exerciseMin13"
                value="13"
              >
                13
              </option>
              <option
                key="exerciseMin14"
                value="14"
              >
                14
              </option>
              <option
                key="exerciseMin15"
                value="15"
              >
                15
              </option>
              <option
                key="exerciseMin16"
                value="16"
              >
                16
              </option>
              <option
                key="exerciseMin17"
                value="17"
              >
                17
              </option>
              <option
                key="exerciseMin18"
                value="18"
              >
                18
              </option>
              <option
                key="exerciseMin19"
                value="19"
              >
                19
              </option>
              <option
                key="exerciseMin20"
                value="20"
              >
                20
              </option>
              <option
                key="exerciseMin21"
                value="21"
              >
                21
              </option>
              <option
                key="exerciseMin22"
                value="22"
              >
                22
              </option>
              <option
                key="exerciseMin23"
                value="23"
              >
                23
              </option>
              <option
                key="exerciseMin24"
                value="24"
              >
                24
              </option>
              <option
                key="exerciseMin25"
                value="25"
              >
                25
              </option>
              <option
                key="exerciseMin26"
                value="26"
              >
                26
              </option>
              <option
                key="exerciseMin27"
                value="27"
              >
                27
              </option>
              <option
                key="exerciseMin28"
                value="28"
              >
                28
              </option>
              <option
                key="exerciseMin29"
                value="29"
              >
                29
              </option>
              <option
                key="exerciseMin30"
                value="30"
              >
                30
              </option>
              <option
                key="exerciseMin31"
                value="31"
              >
                31
              </option>
              <option
                key="exerciseMin32"
                value="32"
              >
                32
              </option>
              <option
                key="exerciseMin33"
                value="33"
              >
                33
              </option>
              <option
                key="exerciseMin34"
                value="34"
              >
                34
              </option>
              <option
                key="exerciseMin35"
                value="35"
              >
                35
              </option>
              <option
                key="exerciseMin36"
                value="36"
              >
                36
              </option>
              <option
                key="exerciseMin37"
                value="37"
              >
                37
              </option>
              <option
                key="exerciseMin38"
                value="38"
              >
                38
              </option>
              <option
                key="exerciseMin39"
                value="39"
              >
                39
              </option>
              <option
                key="exerciseMin40"
                value="40"
              >
                40
              </option>
              <option
                key="exerciseMin41"
                value="41"
              >
                41
              </option>
              <option
                key="exerciseMin42"
                value="42"
              >
                42
              </option>
              <option
                key="exerciseMin43"
                value="43"
              >
                43
              </option>
              <option
                key="exerciseMin44"
                value="44"
              >
                44
              </option>
              <option
                key="exerciseMin45"
                value="45"
              >
                45
              </option>
              <option
                key="exerciseMin46"
                value="46"
              >
                46
              </option>
              <option
                key="exerciseMin47"
                value="47"
              >
                47
              </option>
              <option
                key="exerciseMin48"
                value="48"
              >
                48
              </option>
              <option
                key="exerciseMin49"
                value="49"
              >
                49
              </option>
              <option
                key="exerciseMin50"
                value="50"
              >
                50
              </option>
              <option
                key="exerciseMin51"
                value="51"
              >
                51
              </option>
              <option
                key="exerciseMin52"
                value="52"
              >
                52
              </option>
              <option
                key="exerciseMin53"
                value="53"
              >
                53
              </option>
              <option
                key="exerciseMin54"
                value="54"
              >
                54
              </option>
              <option
                key="exerciseMin55"
                value="55"
              >
                55
              </option>
              <option
                key="exerciseMin56"
                value="56"
              >
                56
              </option>
              <option
                key="exerciseMin57"
                value="57"
              >
                57
              </option>
              <option
                key="exerciseMin58"
                value="58"
              >
                58
              </option>
              <option
                key="exerciseMin59"
                value="59"
              >
                59
              </option>
            </Field>
            :
            <Field
              className="modal--edit__numberInput2"
              component="select"
              max="59"
              name="exerciseSeconds"
              type="text"
            >
              <option
                key="exerciseSec00"
                value="00"
              >
                00
              </option>
              <option
                key="exerciseSec01"
                value="01"
              >
                01
              </option>
              <option
                key="exerciseSec02"
                value="02"
              >
                02
              </option>
              <option
                key="exerciseSec03"
                value="03"
              >
                03
              </option>
              <option
                key="exerciseSec04"
                value="04"
              >
                04
              </option>
              <option
                key="exerciseSec05"
                value="05"
              >
                05
              </option>
              <option
                key="exerciseSec06"
                value="06"
              >
                06
              </option>
              <option
                key="exerciseSec07"
                value="07"
              >
                07
              </option>
              <option
                key="exerciseSec08"
                value="08"
              >
                08
              </option>
              <option
                key="exerciseSec09"
                value="09"
              >
                09
              </option>
              <option
                key="exerciseSec10"
                value="10"
              >
                10
              </option>
              <option
                key="exerciseSec11"
                value="11"
              >
                11
              </option>
              <option
                key="exerciseSec12"
                value="12"
              >
                12
              </option>
              <option
                key="exerciseSec13"
                value="13"
              >
                13
              </option>
              <option
                key="exerciseSec14"
                value="14"
              >
                14
              </option>
              <option
                key="exerciseSec15"
                value="15"
              >
                15
              </option>
              <option
                key="exerciseSec16"
                value="16"
              >
                16
              </option>
              <option
                key="exerciseSec17"
                value="17"
              >
                17
              </option>
              <option
                key="exerciseSec18"
                value="18"
              >
                18
              </option>
              <option
                key="exerciseSec19"
                value="19"
              >
                19
              </option>
              <option
                key="exerciseSec20"
                value="20"
              >
                20
              </option>
              <option
                key="exerciseSec21"
                value="21"
              >
                21
              </option>
              <option
                key="exerciseSec22"
                value="22"
              >
                22
              </option>
              <option
                key="exerciseSec23"
                value="23"
              >
                23
              </option>
              <option
                key="exerciseSec24"
                value="24"
              >
                24
              </option>
              <option
                key="exerciseSec25"
                value="25"
              >
                25
              </option>
              <option
                key="exerciseSec26"
                value="26"
              >
                26
              </option>
              <option
                key="exerciseSec27"
                value="27"
              >
                27
              </option>
              <option
                key="exerciseSec28"
                value="28"
              >
                28
              </option>
              <option
                key="exerciseSec29"
                value="29"
              >
                29
              </option>
              <option
                key="exerciseSec30"
                value="30"
              >
                30
              </option>
              <option
                key="exerciseSec31"
                value="31"
              >
                31
              </option>
              <option
                key="exerciseSec32"
                value="32"
              >
                32
              </option>
              <option
                key="exerciseSec33"
                value="33"
              >
                33
              </option>
              <option
                key="exerciseSec34"
                value="34"
              >
                34
              </option>
              <option
                key="exerciseSec35"
                value="35"
              >
                35
              </option>
              <option
                key="exerciseSec36"
                value="36"
              >
                36
              </option>
              <option
                key="exerciseSec37"
                value="37"
              >
                37
              </option>
              <option
                key="exerciseSec38"
                value="38"
              >
                38
              </option>
              <option
                key="exerciseSec39"
                value="39"
              >
                39
              </option>
              <option
                key="exerciseSec40"
                value="40"
              >
                40
              </option>
              <option
                key="exerciseSec41"
                value="41"
              >
                41
              </option>
              <option
                key="exerciseSec42"
                value="42"
              >
                42
              </option>
              <option
                key="exerciseSec43"
                value="43"
              >
                43
              </option>
              <option
                key="exerciseSec44"
                value="44"
              >
                44
              </option>
              <option
                key="exerciseSec45"
                value="45"
              >
                45
              </option>
              <option
                key="exerciseSec46"
                value="46"
              >
                46
              </option>
              <option
                key="exerciseSec47"
                value="47"
              >
                47
              </option>
              <option
                key="exerciseSec48"
                value="48"
              >
                48
              </option>
              <option
                key="exerciseSec49"
                value="49"
              >
                49
              </option>
              <option
                key="exerciseSec50"
                value="50"
              >
                50
              </option>
              <option
                key="exerciseSec51"
                value="51"
              >
                51
              </option>
              <option
                key="exerciseSec52"
                value="52"
              >
                52
              </option>
              <option
                key="exerciseSec53"
                value="53"
              >
                53
              </option>
              <option
                key="exerciseSec54"
                value="54"
              >
                54
              </option>
              <option
                key="exerciseSec55"
                value="55"
              >
                55
              </option>
              <option
                key="exerciseSec56"
                value="56"
              >
                56
              </option>
              <option
                key="exerciseSec57"
                value="57"
              >
                57
              </option>
              <option
                key="exerciseSec58"
                value="58"
              >
                58
              </option>
              <option
                key="exerciseSec59"
                value="59"
              >
                59
              </option>
            </Field>
          </div>
          <div
            className="exerciseGrid__colorCol"
          >
            <label>
              Color
            </label>
            <Field
              component="select"
              name="exerciseColor"
              style={
                Object {
                  "backgroundColor": "blue",
                }
              }
            >
              <option
                value="#40b3e1"
              />
              <option
                value="#acff82"
              />
              <option
                value="#fedc45"
              />
              <option
                value="#fec345"
              />
              <option
                value="#fe6145"
              />
              <option
                value="#b145fe"
              />
            </Field>
          </div>
        </div>
        <div>
          <label
            className="modal--edit__breakLabel"
          >
            Break
          </label>
        </div>
        <div
          className="modal--edit__checkboxDiv"
        >
          <Field
            className="modal--edit__checkbox"
            id="switch"
            name="breakIncluded"
            type="checkbox"
          />
          <label
            htmlFor="switch"
          >
            Toggle
          </label>
        </div>
        <button
          className="modal__button"
          disabled={true}
        >
          Add Exercise
        </button>
      </Form>
    </div>
  </div>
</Modal>
`;

exports[`should render FormikEditModal correctly 1`] = `
<Formik
  enableReinitialize={false}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "breakColor": "#b145fe",
      "breakIncluded": true,
      "breakMinutes": 0,
      "breakName": "Break",
      "breakSeconds": 0,
      "exerciseColor": "#40b3e1",
      "exerciseMinutes": 0,
      "exerciseName": "",
      "exerciseSeconds": 0,
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={[Function]}
/>
`;
